src/tree_cmd.cpp

```diff
@@ -186,21 +289,28 @@ static void PlaceTree(TileIndex tile, uint32_t r)
  */
 static void PlaceTreeGroups(uint num_groups)
 {
+	static constexpr uint GROVE_SEGMENTS = 16; ///< How many segments make up the tree group.
+	static constexpr uint GROVE_RADIUS = 16; ///< Maximum radius of tree groups.
+	std::array<Point, GROVE_SEGMENTS> grove;
```

Comment:
maybe worth a comment "avoiding allocations" either here or on `CreateRandomStarShapedPolygon` (I assume that's why it's here and not just a return value of the function)